Add linting clarifications (#8700)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/studio-ide/lint-format.md

- Empowers you to [enable linting](#enable-linting) with options like **Lint** (displays linting errors and recommends actions) or **Fix** (auto-fixes errors in the <Constant name="cloud_ide" />).
- Displays a **Code Quality** tab to view code errors, provides code quality visibility and management.

:::info Linting considerations

> - Linting in the <Constant name="cloud_ide" /> currently runs on <Constant name="core" />, even when your development environment uses the **Latest Fusion** release track.

Reviewer: sugg making it clear its the core engine - i dont think everyone knows that 'dbt core' means 'engine' and can infer/assume it's dbt core oss
```suggestion
- The <Constant name="cloud_ide" /> runs linting using the <Constant name="core" /> engine, even when your development environment uses the **Latest Fusion** release track.
```

Reviewer: should we also link to https://docs.getdbt.com/docs/fusion/supported-features#limitations since this is a temp limitation?
